docs/architecture_components.md

```diff
@@ -23,6 +23,8 @@ To suggest a new domain, start a discussion in [the architecture repository](htt
 
 These integrations interact with external devices & services and make them available in Home Assistant via integrations that define IoT domains like `light`.  An example of such an integration is Philips Hue. Philips Hue lights are made available as light entities in Home Assistant.
 
+Integrations which interact with external devices & services are generally not allowed to consume the state of entities from other integrations, with the exception of entities from other integrations which have a location, e.g. the state of zone and device_tracker entities.
```

Comment:
This does not exclude sensor entities with `lat` and `long` set in the state attributes, as they also show on the map. `iss` is an example for this.

When you read this, you could read such entity as "entities from other integrations which have a location"

Reply:
Do we need to assume our developers read the documentation like the devil reads the bible? A sensor, or button, or switch, is not designed to provide a location whereas a device tracker or zone entity is.

Reply:
True true, just sharing this in case you weren't aware. I don't have any other proposals.

Reply:
Your concern is valid, it's just that I also don't have a great idea for how to make it clearer :)

Reply:
We've said that we allow location state from other integrations. It doesn't matter what kind of entity that is. If something seems weird in a PR, we'll take it up for discussion at that time.